subprojects/kotlin-dsl/src/test/kotlin/org/gradle/kotlin/dsl/execution/ProgramParserTest.kt

```diff
@@ -84,14 +91,38 @@ class ProgramParserTest {
 
         val source = ProgramSource(
             "/src/fragment.gradle.kts",
-            "\r\n\r\nplugins {\r\n  java\r\n}\r\nprintln(\"Stage 2\")\r\n\r\n"
+            """
+            // a comment
+            @Suppress("unused_variable")
+
+            /* more comments */
+            @Suppress("whatever")
+
+            plugins {
+              java
+            }
+            @Suppress("nothing_really")
+            println("Stage 2")
+            """.trimIndent()
         )
 
+        val expectedScript = "            \n" +
+            "                            \n" +
+            "\n                   " +
+            "\n                     " +
+            "\n" +
+            "\n         " +
+            "\n      " +
+            "\n " +
+            "\n" +
+            "@Suppress(\"nothing_really\")\n" +
```

Comment:
❓  I think the changes in this file make it less readable. I would prefer we augment `LexerTest` with more coverage as it is actually testing the component responsible for handling annotations around top-level script blocks.  `ProgramParser` on the other hand is responsible for turning a script into a set of instructions and doesn't deal with annotation parsing, it's one level higher. Does that makes sense to you?

Reply:
I don't think the `Lexer` should know about the functionality of erasing some annotations, just as it doesn't know about the extraction of the main blocks. That's the `ProgramParser`'s job. 

I think we should do something else not to make these tests harder to read. Let's discuss that.

Reply:
I agree to your disagreement 👍 

Let's create another test class with test cases specific to annotations in order not to clutter the existing `ProgramParserTest`. This can happen in a follow up PR.